File: tools/_check_zip_aws.js
#!/usr/bin/env node
// 扫描 release zip 内 sdk/aws/Aws 的 Aws\Xxx\ 引用 vs 白名单目录，报告缺失
const fs = require('fs');
const { execSync } = require('child_process');
const path = require('path');

const zipPath = process.argv[2];
if (!zipPath) { console.error('usage: node check_zip_aws.js <zip>'); process.exit(1); }

// 用 python 解出 zip 内 sdk/aws/Aws 的文件清单 + 内容（避免额外依赖）
const script = `
import zipfile, sys
z = zipfile.ZipFile(r'${zipPath.replace(/\\/g, '\\\\')}')
for n in z.namelist():
    if n.startswith('sdk/aws/Aws/') and n.endswith('.php'):
        print('FILE\t' + n)
        print(z.read(n).decode('utf-8', 'ignore'))
`;
const out = execSync(`C:/Users/Administrator/.workbuddy/binaries/python/versions/3.13.12/python.exe -c "${script}"`, { encoding: 'utf8', maxBuffer: 100 * 1024 * 1024 });

const refs = new Set();
let curFile = null;
for (const line of out.split('\n')) {
  if (line.startsWith('FILE\t')) { curFile = line.slice(5); continue; }
  if (curFile) {
    const re2 = /Aws\\([A-Za-z0-9_]+)\\/g;
    let m;
    while ((m = re2.exec(line))) refs.add(m[1]);
  }
}

// 从 zip 拿目录清单（顶层子目录）
const script2 = `
import zipfile
z = zipfile.ZipFile(r'${zipPath.replace(/\\/g, '\\\\')}')
dirs = set()
for n in z.namelist():
    if n.startswith('sdk/aws/Aws/'):
        rest = n[len('sdk/aws/Aws/'):]
        top = rest.split('/')[0]
        if '.' in top: continue  # 文件
        dirs.add(top)
print(' '.join(sorted(dirs)))
`;
const dirsOut = execSync(`C:/Users/Administrator/.workbuddy/binaries/python/versions/3.13.12/python.exe -c "${script2}"`, { encoding: 'utf8' });
const keepDirs = dirsOut.trim().split(/\s+/).filter(Boolean);

const missing = [...refs].filter(ns => !keepDirs.includes(ns) && ns !== 'data');
console.log('zip 内 Aws 子命名空间引用数:', refs.size);
console.log('zip 内保留目录数:', keepDirs.length);
console.log('保留目录:', keepDirs.sort().join(', '));
console.log();
console.log('MISSING（被引用但 zip 里没有）:', missing.length ? missing.join(', ') : 'NONE ✅');

File: tools/archive/scripts/_check_zip_aws.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# 最终验证：剥离注释后扫描最新 v1.0.29 zip 内 sdk/aws/Aws 的真实代码引用 vs 白名单
import zipfile, re, os, glob

zips = sorted(glob.glob(r'D:\projects\2026-08-05-10-13-25\releases\MoeRNG-v1.0.30-*.zip'))
zip_path = zips[-1]
z = zipfile.ZipFile(zip_path)

def strip_comments(src):
    src = re.sub(r'/\*.*?\*/', ' ', src, flags=re.S)
    src = re.sub(r'//[^\n]*', ' ', src)
    src = re.sub(r'#[^\n]*', ' ', src)
    return src

refs = set()
for n in z.namelist():
    if n.startswith('sdk/aws/Aws/') and n.endswith('.php'):
        src = strip_comments(z.read(n).decode('utf-8', 'ignore'))
        for m in re.finditer('Aws\\\\([A-Za-z0-9_]+)\\\\', src):
            refs.add(m.group(1))

keep = set()
for n in z.namelist():
    if n.startswith('sdk/aws/Aws/'):
        top = n[len('sdk/aws/Aws/'):].split('/')[0]
        if '.' not in top:
            keep.add(top)

missing = sorted(r for r in refs if r not in keep and r != 'data')
print('zip:', os.path.basename(zip_path), '| Size: %.2f MB' % (os.path.getsize(zip_path) / 1048576))
print('真实代码引用 (%d): %s' % (len(refs), ', '.join(sorted(refs))))
print('保留目录 (%d): %s' % (len(keep), ', '.join(sorted(keep))))
print('MISSING:', ', '.join(missing) if missing else 'NONE (clean)')
